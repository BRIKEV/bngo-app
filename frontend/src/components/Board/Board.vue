<template>
  <div class="board">
  </div>
</template>

<script>
export default {
  name: 'Board',
};
</script>
