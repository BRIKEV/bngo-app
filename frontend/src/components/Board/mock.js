export default [
  {
    imageUrl: 'https://i.pinimg.com/474x/06/8f/ef/068feff9feec5e0796c5367aaf8cdbd5.jpg',
    caption: 'Photo by 1',
  },
  {
    imageUrl: 'https://i.pinimg.com/originals/87/8a/2c/878a2cb6862f04cefa08fb2c6acedaf6.png',
    caption: 'Photo by 2',
  },
  {
    imageUrl: 'https://i.pinimg.com/originals/31/85/c7/3185c7434b41167a0df2859eb878a158.png',
    caption: 'Photo by 3',
  },
  {
    imageUrl: 'https://i.pinimg.com/originals/90/9d/d6/909dd61d3ea0213e274610c7e658c769.png',
    caption: 'Photo by 4',
  },
  {
    imageUrl: 'https://i.pinimg.com/474x/06/8f/ef/068feff9feec5e0796c5367aaf8cdbd5.jpg',
    caption: 'Photo by 1',
  },
  {
    imageUrl: 'https://i.pinimg.com/originals/87/8a/2c/878a2cb6862f04cefa08fb2c6acedaf6.png',
    caption: 'Photo by 2',
  },
  {
    imageUrl: 'https://i.pinimg.com/originals/31/85/c7/3185c7434b41167a0df2859eb878a158.png',
    caption: 'Photo by 3',
  },
  {
    imageUrl: 'https://i.pinimg.com/originals/90/9d/d6/909dd61d3ea0213e274610c7e658c769.png',
    caption: 'Photo by 4',
  },
  {
    imageUrl: 'https://i.pinimg.com/474x/06/8f/ef/068feff9feec5e0796c5367aaf8cdbd5.jpg',
    caption: 'Photo by 1',
  },
  {
    imageUrl: 'https://i.pinimg.com/originals/87/8a/2c/878a2cb6862f04cefa08fb2c6acedaf6.png',
    caption: 'Photo by 2',
  },
  {
    imageUrl: 'https://i.pinimg.com/originals/31/85/c7/3185c7434b41167a0df2859eb878a158.png',
    caption: 'Photo by 3',
  },
  {
    imageUrl: 'https://i.pinimg.com/originals/90/9d/d6/909dd61d3ea0213e274610c7e658c769.png',
    caption: 'Photo by 4',
  },
  {
    imageUrl: 'https://i.pinimg.com/474x/06/8f/ef/068feff9feec5e0796c5367aaf8cdbd5.jpg',
    caption: 'Photo by 1',
  },
  {
    imageUrl: 'https://i.pinimg.com/originals/87/8a/2c/878a2cb6862f04cefa08fb2c6acedaf6.png',
    caption: 'Photo by 2',
  },
  {
    imageUrl: 'https://i.pinimg.com/originals/31/85/c7/3185c7434b41167a0df2859eb878a158.png',
    caption: 'Photo by 3',
  },
  {
    imageUrl: 'https://i.pinimg.com/originals/90/9d/d6/909dd61d3ea0213e274610c7e658c769.png',
    caption: 'Photo by 4',
  },
];
