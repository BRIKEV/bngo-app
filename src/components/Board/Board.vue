<template>
  <div
    class="board"
    :style="boardStyles(numOfRows, numOfColumns)"
    >
    <div v-for="(item, index) in images" :key="index" class="boardCard">
      <div v-show="!item.selected && !allSelected" class="hiddenCard">?</div>
      <div
        :style="itemStyles(item)"
        :class="{ disabled: !item.selected && !allSelected }"
        class="item" />
    </div>
  </div>
</template>

<script>
import VueTypes from 'vue-types';

export default {
  name: 'Board',
  props: {
    numOfColumns: VueTypes.number.isRequired,
    numOfRows: VueTypes.number.isRequired,
    allSelected: VueTypes.bool.def(false),
    images: VueTypes.arrayOf(VueTypes.shape({
      id: VueTypes.number,
      image: VueTypes.string.isRequired,
      selected: VueTypes.bool.def(false),
    })).loose,
  },
  methods: {
    itemStyles(item) {
      return {
        'background-image': `url(${item.image})`,
      };
    },
    boardStyles(rows, columns) {
      return {
        'grid-template-rows': `repeat(${rows}, auto)`,
        'grid-template-columns': `repeat(${columns}, 1fr)`,
      };
    },
  },
};
</script>
<style lang="scss" scoped>
$boxShadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
$cardRadious: 20px;
.board {
  width: 100%;
  height: 100%;
  display: grid;
  grid-gap: 5px;
  .boardCard {
    position: relative;
    .hiddenCard {
      position: absolute;
      width: 100%;
      height: 100%;
      background: rgba(28, 28, 28, 0.96);
      border-radius: $cardRadious;
      text-align: center;
      color: $white;
      display: flex;
      align-items: center;
      justify-content: center;
      font-size: $fs-h1;
    }
    .item {
      cursor: pointer;
      background-size: cover;
      background-repeat: no-repeat;
      border-radius: $cardRadious;
      box-shadow: $boxShadow;
      transition: transform .3s;
      height: 100%;
      width: 100%;
      &:hover {
        transform: scale(2.5);
        background-position: center;
        z-index: 3;
      }
    }
  }
}
.disabled {
  pointer-events: none;
  position: none;
  cursor: not-allowed;
  transition: opacity .5s;
}
</style>
