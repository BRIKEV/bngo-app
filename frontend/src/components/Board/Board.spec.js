import { mount } from '@vue/test-utils';
import Board from './Board.vue';

describe('Board component', () => {
  const propsData = {
    numOfColumns: 4,
    numOfRows: 4,
  };
  const defaultConfig = {
    propsData,
  };
  it('Should be render correctly', () => {
    const wrapper = mount(Board, defaultConfig);
    expect(wrapper.element).toMatchSnapshot();
  });
});
