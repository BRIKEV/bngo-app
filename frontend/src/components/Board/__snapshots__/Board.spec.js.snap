// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component Should be render correctly 1`] = `
<div
  class="board"
  style="grid-template-rows: repeat(4, auto); grid-template-columns: repeat(4, 1fr);"
/>
`;
